--- origsrc/glib-2.16.5/tests/module-test.c	2008-07-01 17:30:08.000000000 -0500
+++ src/glib-2.16.5/tests/module-test.c	2008-07-22 18:05:16.328125000 -0500
@@ -85,9 +85,15 @@
 
   dir = g_get_current_dir ();
 
-  plugin_a = g_strconcat (dir, G_DIR_SEPARATOR_S "libmoduletestplugin_a", 
+#ifdef G_WITH_CYGWIN
+#define PLUGIN_NAME ".libs/cygmoduletestplugin"
+#else
+#define PLUGIN_NAME "libmoduletestplugin"
+#endif
+
+  plugin_a = g_strconcat (dir, G_DIR_SEPARATOR_S PLUGIN_NAME "_a", 
 			  NULL);
-  plugin_b = g_strconcat (dir, G_DIR_SEPARATOR_S "libmoduletestplugin_b", 
+  plugin_b = g_strconcat (dir, G_DIR_SEPARATOR_S PLUGIN_NAME "_b", 
 			  NULL);
 
   g_free (dir);
@@ -96,38 +102,40 @@
   
   module_self = g_module_open (NULL, G_MODULE_BIND_LAZY);
   if (!module_self)
-    g_error ("error: %s", g_module_error ());
+    g_error ("error1: %s", g_module_error ());
 
+#ifndef G_WITH_CYGWIN
   if (!g_module_symbol (module_self, "g_module_close", (gpointer *) &f_self))
-    g_error ("error: %s", g_module_error ());
+    g_error ("error2: %s", g_module_error ());
+#endif
 
   module_a = g_module_open (plugin_a, G_MODULE_BIND_LAZY);
   if (!module_a)
-    g_error ("error: %s", g_module_error ());
+    g_error ("error3: %s", g_module_error ());
 
   module_b = g_module_open (plugin_b, G_MODULE_BIND_LAZY);
   if (!module_b)
-    g_error ("error: %s", g_module_error ());
+    g_error ("error4: %s", g_module_error ());
 
   /* get plugin state vars */
 
   if (!g_module_symbol (module_a, "gplugin_a_state", 
 			(gpointer *) &gplugin_a_state))
-    g_error ("error: %s", g_module_error ());
+    g_error ("error5: %s", g_module_error ());
   
   if (!g_module_symbol (module_b, "gplugin_b_state", 
 			(gpointer *) &gplugin_b_state))
-    g_error ("error: %s", g_module_error ());
+    g_error ("error6: %s", g_module_error ());
   test_states (NULL, NULL, "check-init");
   
   /* get plugin specific symbols and call them
    */
   if (!g_module_symbol (module_a, "gplugin_a_func", (gpointer *) &f_a))
-    g_error ("error: %s", g_module_error ());
+    g_error ("error7: %s", g_module_error ());
   test_states (NULL, NULL, NULL);
  
   if (!g_module_symbol (module_b, "gplugin_b_func", (gpointer *) &f_b))
-    g_error ("error: %s", g_module_error ());
+    g_error ("error8: %s", g_module_error ());
   test_states (NULL, NULL, NULL);
  
   f_a ();
@@ -140,15 +148,15 @@
    */
  
   if (!g_module_symbol (module_self, "g_clash_func", (gpointer *) &f_self))
-    g_error ("error: %s", g_module_error ());
+    g_error ("error9: %s", g_module_error ());
   test_states (NULL, NULL, NULL);
 
   if (!g_module_symbol (module_a, "g_clash_func", (gpointer *) &f_a))
-    g_error ("error: %s", g_module_error ());
+    g_error ("error10: %s", g_module_error ());
   test_states (NULL, NULL, NULL);
  
   if (!g_module_symbol (module_b, "g_clash_func", (gpointer *) &f_b))
-    g_error ("error: %s", g_module_error ());
+    g_error ("error11: %s", g_module_error ());
   test_states (NULL, NULL, NULL);
  
   f_self ();
@@ -163,11 +171,11 @@
   /* get and call clashing plugin functions  */
 
   if (!g_module_symbol (module_a, "gplugin_clash_func", (gpointer *) &f_a))
-    g_error ("error: %s", g_module_error ());
+    g_error ("error12: %s", g_module_error ());
   test_states (NULL, NULL, NULL);
 
   if (!g_module_symbol (module_b, "gplugin_clash_func", (gpointer *) &f_b))
-    g_error ("error: %s", g_module_error ());
+    g_error ("error13: %s", g_module_error ());
   test_states (NULL, NULL, NULL);
 
   plugin_clash_func = f_a;
@@ -181,7 +189,7 @@
   /* call gmodule function from A  */
 
   if (!g_module_symbol (module_a, "gplugin_a_module_func", (gpointer *) &gmod_f))
-    g_error ("error: %s", g_module_error ());
+    g_error ("error14: %s", g_module_error ());
   test_states (NULL, NULL, NULL);
 
   gmod_f (module_b);
@@ -193,10 +201,10 @@
   /* unload plugins  */
 
   if (!g_module_close (module_a))
-    g_error ("error: %s", g_module_error ());
+    g_error ("error15: %s", g_module_error ());
 
   if (!g_module_close (module_b))
-    g_error ("error: %s", g_module_error ());
+    g_error ("error16: %s", g_module_error ());
  
   return 0;
 }
