--- origsrc/glib-2.36.3/gio/gunixmounts.c	2013-06-09 17:03:17.000000000 -0500
+++ src/glib-2.36.3/gio/gunixmounts.c	2013-08-04 18:47:53.856023300 -0500
@@ -774,9 +774,11 @@ _g_get_unix_mount_points (void)
           (strcmp (mntent->mnt_dir, "none") == 0))
 	continue;
 
+#ifdef HAVE_HASMNTOPT
       /* We ignore bind fstab entries, as we ignore bind mounts anyway */
       if (hasmntopt (mntent, "bind"))
         continue;
+#endif
 
       mount_entry = g_new0 (GUnixMountPoint, 1);
       mount_entry->mount_path = g_strdup (mntent->mnt_dir);
